lib/processors/minifier.js

```diff
@@ -29,10 +29,24 @@ function getPool(taskUtil) {
 			maxWorkers
 		});
 		taskUtil.registerCleanupTask(() => {
-			log.verbose(`Terminating workerpool`);
-			const poolToBeTerminated = pool;
-			pool = null;
-			poolToBeTerminated.terminate();
```

Comment:
According to the [documentation](https://github.com/josdejong/workerpool#pool) `Pool.terminate` supposedly takes care of the graceful termination of workers that are still busy:

> If parameter `force` is false (default), workers will finish the tasks they are working on before terminating themselves. Any pending tasks will be rejected with an error 'Pool terminated'.

I actually wasn't aware that `Pool.terminate` returns a promise. Apparently the examples in the [README](https://github.com/josdejong/workerpool#use) don't wait for it either. But maybe that's the main fix here? Make the callback function async (taskUtil will wait for any returned promise) and  `await poolToBeTerminated.terminate();`?

Reply:
Thank you, @RandomByte!
Basically, it would make the issue harder to reproduce, but it would still be there.
Here the key point is `tasks they are working on`. 
Your proposal makes a lot of sense IF `busy workers` === `total tasks`.

Please take a look at the tests cases with this change. I intentionally made them with more data to expose the real problem. We need to wait until there are no busy, pending nor active tasks in the queue. Awaiting pool's termination would wait for the active tasks to finnish, but all of the pending tasks would be cancelled.

Reply:
Indeed I missed how workerpool handles queued task during its termination. Since it **[rejects]()** all queued tasks, we end up with `Pool terminated` issue in scenarios like https://github.com/SAP/ui5-tooling/issues/894

So this change fixes that case by letting all queued tasks finish executing before terminating the pool.

However, this means when *cancelling* a UI5 CLI build (e.g. using CTRL+C), the pool will also wait for all queued tasks to finish before terminating. This can be noticed for example when building the `sap.m` library and cancelling the build right after the `minify` task starts. With this PR it takes ~2 seconds for ending the process. Before it would end almost immediately

Reply:
Thanks for finding it out!
We'll need to have the execution contex in that case. Indeed, it's quite slow, when we forcefully try to terminate it. In this case it should enforce the termination.

I have created a complimentary change that provides this context: https://github.com/SAP/ui5-project/pull/677